client/src/pages/AssignMates/AssignMatesPage.tsx

```diff
@@ -0,0 +1,221 @@
+import { Button, Col, Divider, Layout, Row, Space, Typography } from "antd";
+import React, { useState } from "react";
+import {
+  DragDropContext,
+  DraggableLocation,
+  DropResult,
+} from "react-beautiful-dnd";
+import { Prompt } from "react-router";
+
+import FamilyDroppable from "../../components/FamilyDroppable/FamilyDroppable";
+import { NewMateType } from "../types";
+
+const { Title } = Typography;
+const { Footer } = Layout;
+
+// hardcoded data for demo
+const philosophers = [
+  "Thales",
+  "Democritus",
+  "Empedokles",
+  "Pythagoras",
+  "Socrates",
+  "Plato",
+  "Aristotle",
+  "Diogenes",
+  "Anaxagoras",
+  "Euclides",
+  "Antisthenes",
+  "Epicurus",
+  "Zeno of Citium",
+];
+const initialMates = philosophers.map(
+  (name, index) =>
+    ({
+      name,
+      lastSynced: new Date(),
+      id: `mate-id-${index}`,
+    } as NewMateType)
+);
+const families = ["school", "work", "life", "college", "concert"];
+const initialGroup: Record<string, NewMateType[]> = families.reduce(
+  (object, family) => ({
+    ...object,
+    [family]: [],
+  }),
+  { unassigned: initialMates }
+);
+// hardcoded data for demo
+
+/**
+ * Reorder an item in a list
+ * @param list
+ * @param startIndex original index of item
+ * @param endIndex new index of item
+ * @returns reordered list
+ */
+const reorder = <T,>(list: T[], startIndex: number, endIndex: number): T[] => {
```

Comment:
Wasn't sure if it was better to leave it here or refactor it to helper function file since these are specific to this page.

Reply:
Imo everything stays in same class until another class wants to also use it - then you move it out.